frontend/helpers/product-list-helpers.ts

```diff
@@ -57,16 +57,35 @@ export function stylizeDeviceItemType(itemType: string): string {
 
 /**
  * Convert '_' in URL slug to spaces for product list device title
+ * If the handle has a : character everything after it is removed.
+ * This is to remove the device's variant from the handle.
+ * Example: "PlayStation_4:CUH-11XXA" becomes "PlayStation 4"
  */
 export function destylizeDeviceTitle(handle: string): string {
-   return handle.replace(/_/g, ' ');
+   const [device, _variant] = handle.split(':');
+   return device.replace(/_/g, ' ');
+}
+
+/**
+ * Keeping the device's variant in the handle.
+ * Convert '_' in URL slug to spaces for product list device title
+ * Convert ':' in URL slug to spaces for the variant separator
+ * Example: "PlayStation_4:CUH-11XXA" becomes "PlayStation 4 CUH-11XXA"
+ */
+export function destylizeDeviceTitleAndVariant(handle: string): string {
+   return handle.replace(/_|:/g, ' ');
 }
 
 /**
  * Convert to spaces in product list device Title to '_' for URL slug
  */
-export function stylizeDeviceTitle(deviceTitle: string): string {
-   return deviceTitle.replace(/\s/g, '_');
+export function stylizeDeviceTitle(
+   deviceTitle: string,
+   variant?: string
+): string {
+   const urlTitle = deviceTitle.replace(/\s/g, '_');
+   const urlVariant = variant ? `:${variant}` : '';
+   return urlTitle + urlVariant;
 }
```

Comment:
Will the variant ever have spaces? If so, do we want those spaces to be converted to `_`?

Reply:
Ah, good point. I am having a hard time finding works-in which would have spaces in it.

Take this example:

![image](https://github.com/iFixit/react-commerce/assets/857362/5d2c5c34-a7c6-4a17-98c9-e1402c15b1a8)

`Google Pixel 7 Pro GP4BC` would be the works in, but the base page is `/Parts/Google_Pixel_7_Pro` This means the URL will end up `/Parts/Google_Pixel_7_Pro:GP4BC`

Give me some time to dig into our data and I'll have a concrete answer

Reply:
Spaces exist in the topic_variants table, but I'm not sure where if that's where the variant title comes from

```
mysql> select variant from topic_variants where variant like "% %" limit 20;
+--------------------------------------+
| variant                              |
+--------------------------------------+
| A1181 (EMC 2092 MacBook1,1) 1.83 GHz |
| A1181 (EMC 2092 MacBook1,1) 2 GHz    |
| A1051 1st gen 4 GB                   |
| A1051 2nd gen 4 GB                   |
| A1051 2nd gen 6 GB                   |
| A1241 16GB                           |
| A1241 8GB                            |
| A1137 1GB                            |
| A1137 2GB                            |
| A1137 4GB                            |
| A1199 2GB                            |
| A1199 4GB                            |
| A1199 8GB                            |
| A1236 4GB                            |
| A1236 8GB                            |
| A1285 16GB                           |
| A1285 8GB                            |
| A1112 1 GB                           |
| A1112 512 MB                         |
| A1204 1 GB                           |
+--------------------------------------+
20 rows in set (0.00 sec)
```

Reply:
Spaces exist in the topic_variants table, but I'm not sure where if that's where the variant title comes from

```sql
select variant from topic_variants where variant like "% %"
```

Reply:
> I'm not sure where if that's where the variant title comes from

You can be sure. You are correct:

https://github.com/iFixit/ifixit/blob/d628373ca823190783ad8909cbc0b3aacd29f04c/Objects/Search/Algolia/Handlers/AlgoliaProductGroupHandler.php#L169-L186

18% of variants have spaces:
![image](https://github.com/iFixit/react-commerce/assets/26855/9f684404-6532-4a05-8bf8-721d44e4f757)